pkg/http/api.go

```diff
@@ -780,27 +781,35 @@ func (a *api) onDirectMessage(reqCtx *fasthttp.RequestCtx) {
 	respond(reqCtx, statusCode, body)
 }
 
-func (a *api) onCreateActorReminder(reqCtx *fasthttp.RequestCtx) {
+func (a *api) getActorWithRequestValidation(reqCtx *fasthttp.RequestCtx, req interface{}) (string, string, error) {
+	var err error
 	if a.actor == nil {
 		msg := NewErrorResponse("ERR_ACTOR_RUNTIME_NOT_FOUND", messages.ErrActorRuntimeNotFound)
 		respondWithError(reqCtx, fasthttp.StatusInternalServerError, msg)
-		return
+		return "", "", errors.New(msg.Message)
 	}
 
 	actorType := reqCtx.UserValue(actorTypeParam).(string)
 	actorID := reqCtx.UserValue(actorIDParam).(string)
-	name := reqCtx.UserValue(nameParam).(string)
+	if req != nil && !reflect.ValueOf(req).IsNil() {
```

Comment:
Is `reflect.ValueOf(req).IsNil()` needed? It might be redundant in terms of validation logic and costly in terms of performance.

Reply:
nil and (*string)(nil) are different. Using reflect checking is more strict.

But in the dapr runtime inside it seems unnecessarily.